pageserver/src/tenant/timeline/layer_manager.rs

```diff
@@ -223,28 +240,26 @@ impl LayerManager {
     /// Called when garbage collect the timeline. Returns a guard that will apply the updates to the layer map.
     pub(crate) fn finish_gc_timeline(
         &mut self,
-        layer_removal_cs: Arc<tokio::sync::OwnedMutexGuard<()>>,
-        gc_layers: Vec<Arc<dyn PersistentLayer>>,
-        metrics: &TimelineMetrics,
+        layer_removal_cs: &Arc<tokio::sync::OwnedMutexGuard<()>>,
+        gc_layers: Vec<Layer>,
     ) -> Result<ApplyGcResultGuard> {
         let mut updates = self.layer_map.batch_update();
         for doomed_layer in gc_layers {
             Self::delete_historic_layer(
-                layer_removal_cs.clone(),
+                layer_removal_cs,
                 doomed_layer,
                 &mut updates,
-                metrics,
                 &mut self.layer_fmgr,
-            )?; // FIXME: schedule succeeded deletions in timeline.rs `gc_timeline` instead of in batch?
+            )?;
         }
         Ok(ApplyGcResultGuard(updates))
     }
 
     /// Helper function to insert a layer into the layer map and file manager.
     fn insert_historic_layer(
-        layer: Arc<dyn PersistentLayer>,
+        layer: Layer,
         updates: &mut BatchedUpdates<'_>,
-        mapping: &mut LayerFileManager,
+        mapping: &mut LayerFileManager<Layer>,
     ) {
```

Comment:
All callers except timeline load procedure to

```
 l.access_stats().record_residence_event(
                LayerResidenceStatus::Resident,
                LayerResidenceEventReason::LayerCreate,
            );
            metrics
                .resident_physical_size_gauge
                .add(l.layer_desc().file_size);
```

Reply:
Or, why do we increment the physical size gauge here instead of in the Layer constructor-like functions.

Reply:
Yeah, a lot stuff needs to be fixed with the LayerManager still. I've already straighten some aspects and reduced cloning and whatnot on this PR.

I am a bit vague what do you think these are:
> Layer constructor-like functions

Because no metrics are modified in `Layer::for_{evicted,resident,tempfile}`, only in eviction, download and garbage collection.

Reply:
Physical total is not so easy to move to layer_manager. I think I would rather not do that.

Still interested on the "Layer constructor-like functions".

Reply:
> Because no metrics are modified in Layer::for_{evicted,resident,tempfile}, only in eviction, download and garbage collection.

But, the timeline load code that uses `for_evicted` / `for_resident` does increment the metric.

And similarly, the `LayerAccessStats::for_loading_layer` is also only called by `for_evicted` and `for_resident`.

Reply:
Ok, I failed to catch you wished the metrics were updated within the for_resident and possibly the renaming one. Okay, makes sense.

Reply:
For created layers, the residence events were recorded double.

Reply:
774f34d77